apps/web/utils/logic/variableItem.ts

```diff
@@ -77,19 +113,191 @@ export const getVariableItemValueObject = (
   return newObjectVariable;
 };
 
-export const answerValuesAreNotEqual = (
-  newAnswerValue: AnswerTypes,
-  currentAnswerValue: AnswerTypes,
-): boolean => {
-  if (isArray(newAnswerValue) && isArray(currentAnswerValue)) {
-    return !newAnswerValue.every((value) => currentAnswerValue.includes(value));
-  } else if (isString(newAnswerValue) && isString(currentAnswerValue)) {
-    return newAnswerValue !== currentAnswerValue;
-  } else if (isObject(newAnswerValue) && isObject(currentAnswerValue)) {
-    return !Object.entries(newAnswerValue).every(
-      ([key, value]) => currentAnswerValue[key] === value,
+export const getVariableItemValueNumber = (
+  variableToSet: VariableToSet,
+  getAnswerToCheckValue: AnswerToCheckValueFn,
+): number => {
+  if (isNumber(variableToSet.variableValue)) {
+    return variableToSet.variableValue;
+  } else if (isArray(variableToSet.variableValue)) {
+    for (const variableValueLogicItem of variableToSet.variableValue) {
+      if (
+        ThisModule.variableValueLogicItemIsTrue(
+          variableValueLogicItem,
+          getAnswerToCheckValue,
+        )
+      ) {
+        if (variableValueLogicItem.variableValueToSetCalculation) {
+          const answerToUseValue = getAnswerToCheckValue(
+            variableValueLogicItem.variableValueToSetCalculation.answerToUse,
+          );
+
+          if (answerToUseValue && isNumber(answerToUseValue)) {
+            return calculateVariableValueToSet(
+              answerToUseValue,
+              variableValueLogicItem.variableValueToSetCalculation,
+            );
+          }
+
+          throw new Error(
+            "variableValueToSetCalculation: No answerToUseValue found, or it was not a number.",
+          );
+        }
+        if (isNumber(variableValueLogicItem.variableValueToSet)) {
+          return variableValueLogicItem.variableValueToSet;
+        }
+
+        throw new Error(
+          "variableValueLogicItem: must have variableValueToSetCalculation or variableValueToSet as a number.",
+        );
+      }
+    }
+  }
+
+  throw new Error(
+    `getVariableItemValueString: variableToSet.variableValue must be a number, or array, got ${typeof variableToSet.variableValue}`,
+  );
+};
+
+export const variableValueLogicItemIsTrue = (
+  variableValueLogicItem: VariableValueLogic,
+  getAnswerToCheckValue: AnswerToCheckValueFn,
+) => {
+  // if fallback item is found, set variableValue to fallback item
```

Comment:
```suggestion
) => {
  const {
    variableValueLogicType,
    valuesToCheck,
    answerToCheck,
    answerValue,
    answerValues,
  } = variableValueLogicItem;
  // if fallback item is found, set variableValue to fallback item
```

This would benefit a lot by deconstructing here and some other places.

Reply:
Refactored! I tried to make it JIT so we're not de-structuring before we need to. LMK what you think!